### ISSUE TITLE
Need to use template variables in OpenTSDB filter regexp's

### ISSUE THREAD

**Issue by dominicnicholas** [grafana/grafana#4831]
Hi,

Is it possible to use a template variable in an OpenTSDB filter regular expression please ? If not, please can this be a feature request to expand template variables first in regex string, and then use the resulting string as the regex.

Example:

- template variable called 'hg' of type Custom with possible values 'hg1' and 'hg2'
- a graph with metric filter like this : hostgroups = regexp(^.*unixhosts.*$) , groupBy = true
- need to do this instead : hostgroups = regexp(^.*$hg.*$) , groupBy = true

Version info:  Grafana 3 beta 5 / OpenTSDB 2.2 / CentOS 6.7

Thanks!

**Comment by bobrik** [comment]
@torkelo it seems like Grafana lost an ability to use variables in OpenTSDB filters:

![image](https://cloud.githubusercontent.com/assets/89186/15012124/078f048c-11ee-11e6-9b0d-860252e745fc.png)

OpenTSDB complains that there is not tagv `$node`.

**Comment by utkarshcmu** [comment]
This is the PR which can resolve this issue : https://github.com/grafana/grafana/pull/4594

**Comment by torkelo** [comment]
merged

**Comment by utkarshcmu** [comment]
Thanks @torkelo 

@bobrik Template variables should work in filters now.

**Comment by bobrik** [comment]
@utkarshcmu just checked 3.0.1 and it's still broken:

```json
{
  "start": 1463110424372,
  "queries": [
    {
      "metric": "cadvisor.container_cpu_system_seconds_total",
      "aggregator": "sum",
      "rate": true,
      "rateOptions": {
        "counter": true,
        "resetValue": 1
      },
      "downsample": "1m-avg-null",
      "filters": [
        {
          "type": "literal_or",
          "tagk": "marathon_app_id",
          "filter": "$marathon_app_id",
          "groupBy": false
        }
      ]
    }
  ],
  "msResolution": false,
  "globalAnnotations": true
}
```

![image](https://cloud.githubusercontent.com/assets/89186/15253093/9b3f91ba-1928-11e6-9eb9-ff3cb08ffe97.png)

**Comment by utkarshcmu** [comment]
@bobrik will look into this today.

**Comment by utkarshcmu** [comment]
@bobrik Able to replicate the issue, looks like if "group by" is true then template variables work. But when its false, then they do not work. Can you try running the same query by setting "group by" as true and verify my claim?

Thanks for reporting this issue. Will try to fix it asap.

**Comment by bobrik** [comment]
@utkarshcmu correct, it works with "group by". Thanks for looking into it.

**Comment by utkarshcmu** [comment]
@bobrik - "group by = false" against Opentsdb 2.3RC1 fails - 

https://github.com/OpenTSDB/opentsdb/issues/794 

Will try to fix this atleast for Opentsdb 2.2.0 over the weekend.

**Comment by utkarshcmu** [comment]
Actually, there is nothing to fix here I guess, because when "group by" is false, its only metric which is returned from Opentsdb and no information on tags is returned. So, $tag_anything won't work in the alias as there is no tag returned with the metric.
